### PR TITLE
Add new `Style/EmptyMethod` cop

### PR DESCRIPTION
**Pull request by Drenmi**
This cop checks for the style of empty method definitions. By default it enforces a compact style, i.e.:

```
# EnforcedStyle: compact
def foo; end
```

but it can be configured to enforce a loose style as well, i.e.:

```
# EnforcedStyle: expanded
def foo
end
```

The cop comes with auto-correct.

-----------------

Before submitting the PR make sure the following are checked:

* [X] Wrote [good commit messages][1].
* [X] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [X] Used the same coding conventions as the rest of the project.
* [X] Feature branch is up-to-date with `master` (if not - rebase it).
* [X] Squashed related commits together.
* [X] Added tests.
* [X] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [X] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [X] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [X] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -447,6 +447,12 @@ Style/EmptyLinesAroundModuleBody:
     - empty_lines_special
     - no_empty_lines
 
+Style/EmptyMethod:
+  EnforcedStyle: compact
+  SupportedStyles:
+    - compact
+    - loose
```
Comment (bbatsov) [inline config/default.yml]: I think I prefer "expanded" or something like this.

File: lib/rubocop/cop/style/character_literal.rb [hunk 2]
```diff
@@ -32,13 +32,11 @@ def autocorrect(node)
 
         # Dummy implementation of method in ConfigurableEnforcedStyle that is
         # called from StringHelp.
-        def opposite_style_detected
-        end
+        def opposite_style_detected; end
```
Comment (bbatsov) [inline lib/rubocop/cop/style/character_literal.rb]: By the way `;` is not really needed here. Maybe we should have two styles for enforcing its presence or absence.

Comment (Drenmi) [inline lib/rubocop/cop/style/character_literal.rb]: Hm. I think you might be confusing the first `;` in the definition and the (optional) second one when there's a method body, i.e.:

```
def foo; bar end
```

Defining without the `;` will raise:

```
SyntaxError: (irb):1: syntax error, unexpected keyword_end, expecting ';' or '\n'
```

😊

File: config/default.yml (hunk 1 above)
Comment (Drenmi) [inline config/default.yml]: Fixed. 😀

File: lib/rubocop/cop/style/character_literal.rb (hunk 2 above)
Comment (bbatsov) [inline lib/rubocop/cop/style/character_literal.rb]: Guess I am. :-)

**Comment by lucascaton** [comment]
I'm pretty sure most developers will agree it should defaults to `expanded`.
Also: https://github.com/bbatsov/ruby-style-guide#no-single-line-methods.

Cheers.

**Comment by Drenmi** [comment]
@lucascaton: The best way to test is to open up an issue or a PR in the style guide repository to see if people agree. 😀

For now, the section you linked (and that is also linked in the cop itself) states:

> One exception to the rule are empty-body methods.
>
> ```
> # good
> def no_op; end
> ```

**Pull request merged**